server_test.go

```diff
@@ -2857,6 +2857,43 @@ func TestShutdownDone(t *testing.T) {
 	verifyResponse(t, br, StatusOK, "aaa/bbb", "real response")
 }
 
+func TestShutdownErr(t *testing.T) {
+	ln := fasthttputil.NewInmemoryListener()
+	s := &Server{
+		Handler: func(ctx *RequestCtx) {
+			// This will panic, but I was not able to intercept with recover()
```

Comment:
why?

Reply:
The origin of the panic is here: `/usr/local/go/src/context/context.go:259`

```go
241 // propagateCancel arranges for child to be canceled when parent is.
242 func propagateCancel(parent Context, child canceler) {
...
259     go func() {
260       select {
261       case <-parent.Done():
262         child.cancel(false, parent.Err())
263       case <-child.Done():
264       }
265     }()
266   }
267 }
```
this in some point will call:

```go
346 // cancel closes c.done, cancels each of c's children, and, if
347 // removeFromParent is true, removes c from its parent's children.
348 func (c *cancelCtx) cancel(removeFromParent bool, err error) {
349   if err == nil {
350     panic("context: internal error: missing cancel error")
351   }
```

but since it happens in a goroutine started on `propagateCancel`, I had no success in add a `defer` with `recover` the panic.